include/flatbuffers/verifier.h

```diff
@@ -25,22 +25,48 @@ namespace flatbuffers {
 // Helper class to verify the integrity of a FlatBuffer
 class Verifier FLATBUFFERS_FINAL_CLASS {
  public:
-  Verifier(const uint8_t *const buf, const size_t buf_len,
-           const uoffset_t _max_depth = 64,
-           const uoffset_t _max_tables = 1000000,
-           const bool _check_alignment = true)
+  struct Options {
+    // The maximum nesting of tables and vectors before we call it invalid.
+    uoffset_t max_depth = 64;
+    // The maximum number of tables we will verify before we call it invalid.
+    uoffset_t max_tables = 1000000;
+    // If true, verify all data is aligned.
+    bool check_alignment = true;
+    // If true, run verifier on nested flatbuffers
+    bool check_nested_flatbuffers = true;
+
+    // Manually implemented constructors for compilers that can't figure it out.
+    Options() = default;
+    Options(uoffset_t d, uoffset_t t, bool a, bool n)
+        : max_depth(d),
+          max_tables(t),
+          check_alignment(a),
+          check_nested_flatbuffers(n) {}
+  };
+
+
+  Verifier(const uint8_t *const buf, const size_t buf_len, const Options opts)
```

Comment:
I still feel like passing by ref is preferred, even if the struct is very basic.